examples/ags/w_action_node/graph.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# @Date    : 6/27/2024 22:07 PM
+# @Author  : didi
+# @Desc    : graph & an instance - humanevalgraph
+
+from metagpt.llm import LLM 
+from typing import List
+from examples.ags.w_action_node.operator import Generate, GenerateCode, GenerateCodeBlock, Review, Revise, FuEnsemble, MdEnsemble, DbEnsemble, Rephrase, Test
+from examples.ags.w_action_node.utils import extract_test_cases_from_jsonl
+from evalplus.data import get_human_eval_plus
+class Graph:
+    def __init__(self, name:str, llm:LLM) -> None:
+        self.name = name
+        self.model = llm 
+
+    def __call__():
+        NotImplementedError("Subclasses must implement __call__ method")
+
+    def optimize(dataset:List):
+        pass
+
+class HumanEvalGraph(Graph):
```

Comment:
seems graph has no abstraction, each `{Dataset}Graph` should define the particular operator?